dotnet/src/Planners/Planners.Core/Handlebars/HandlebarsTemplateEngineExtensions.cs

```diff
@@ -0,0 +1,377 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using HandlebarsDotNet;
+using HandlebarsDotNet.Compiler;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.Text;
+
+namespace Microsoft.SemanticKernel.Planners.Handlebars;
+
+/// <summary>
+/// Provides extension methods for rendering Handlebars templates in the context of a Semantic Kernel.
+/// </summary>
+public static class HandlebarsTemplateEngineExtensions
```

Comment:
@markwallace-microsoft I was hoping to integrate this with the TemplateEnginer.Handlebars you created but we need more support for registering the built in helpers. I'll create an ADR for this.

Reply:
Can this be made internal?